## 두 모델에 대한 앙상블 모델 구하기
### 검증데이터에 예측값과 실제값과의 오차에 대한 분산값 이용
### 분산을 역수로 취해주며 값이 높을수록 높은 가중치를 부여함
### 해당 방법은 분산이 작을수록 모델적합도가 높다는 가정하에 진행함

In [56]:
import pandas as pd
import numpy as np
LR = pd.read_csv('data/LinearRegression.csv', delimiter=',')
RF = pd.read_csv('data/RandomForest.csv', delimiter=',')
test = pd.read_csv('data/test.csv', delimiter=',')

### 검증데이터에 예측값과 실제값과의 오차에 대한 분산값
### 해당값을 역수로 취해줌(값이 높을수록 가중치를 높게 부여하기 위함)

In [57]:
alpha=1/12.78 ## 선형회귀
beta=1/18.47 ## 랜덤포레스트

In [58]:
ensemble = (alpha*LR.iloc[:,0]+ beta*RF.iloc[:,0])/(alpha+beta)

In [59]:
np.sqrt(np.mean((ensemble - test.iloc[:,-1])**2))

4.473125207783047

## 최종결과(MSE)
### 앙상블 모델: 4.47
### 회귀 모델 : 4.89
### 랜덤포레스트 모델 : 4.54

### 앙상블 모델을 이용하여 기존 모델보다 더 정확한 모델 생성